File: OrienGO-Backend/src/main/java/com/example/oriengo/model/dto/QuestionResponseDTO.java
package com.example.oriengo.model.dto;

import com.example.oriengo.model.enumeration.Category;
import lombok.*;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@ToString
public class QuestionResponseDTO {

    private Long id;

    private Category category;

    private String text;

    // Include answer options in the response DTO, but without circular references or lazy-loading issues
    private Set<AnswerOptionResponseDTO> answerOptions;

}

File: OrienGO-Frontend/src/app/model/dto/QuestionResponse.dto.ts
import { Category } from '../enum/Category.enum';
import { AnswerOptionResponseDTO } from './AnswerOptionResponse.dto';

export interface QuestionResponseDTO {
  category?: Category;
  id: number;
  text: string;
  answerOptions: AnswerOptionResponseDTO[];
  softDeleted?: boolean
}
